apollo-federation/src/operation/mod.rs

```diff
@@ -2683,16 +2707,11 @@ impl SelectionSet {
                 return Ok(updated.into());
             };
             // We need to add the query __typename for the current type in the current group.
-            // Note that the value of the sibling_typename is the alias or "" if there is no alias
-            let alias = if sibling_typename.is_empty() {
-                None
-            } else {
-                Some(sibling_typename.clone())
-            };
+            let alias = sibling_typename.alias();
             let field_element = Field::new_introspection_typename(
                 &self.schema,
                 &selection.element()?.parent_type_position(),
-                alias,
+                alias.cloned(),
```

Comment:
```suggestion
            sibling_typename.alias().cloned(),
```
We don't really need the `alias` binding here anymore